include/superior_mysqlpp/extras/member_wrapper.hpp

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+namespace SuperiorMySqlpp
+{
+    namespace
+    {
+        /**
+         * @brief Wraps member object and member function pointer into single functor
+         * @tparam T Wrapped class type
+         * @tparam TMember Type of member function
+         * @tparam Args Arguments of TMember function
+         */
+        template<typename T, typename TMember, typename... Args>
+        struct MemberWrapper
+        {
+            T *object;
+            TMember member;
+
+            /**
+             * @brief Constructs MemberWrapper object
+             * @param object Instance of T
+             * @param member Member function from T
+             */
+            constexpr MemberWrapper(T *object, TMember member) noexcept
+                : object(object), member(member) {}
+
+            /**
+             * @brief Invokes stored member function with args
+             * @param args Parameter pack of function arguments
+             */
+            inline void operator()(Args... args) const
+            {
+                return (object->*member)(args...);
+            }
+        };
+    }
+
+    /**
+     * @brief Constructs MemberWrapper object from pointer to T and member function
+     *        returning void
+     * @param object pointer to T
+     * @param member function from T
+     */
+    template<typename T, typename... Args>
+    inline auto wrapMember(T *object, void (T::*member)(Args...)) noexcept
```

Comment:
Maybe non-`void` returning functions could be also supported (+ overloads for _rvalue references for *this_ ?).

Reply:
For now, we can't do anything with result value. It would be ignored, so thats why there is void.
But there is plan for future to:
1) accept function with bool result, that controls, whether result fetch loop will continue or will be broken.
2) accept function with arbitrary return value and another reduce function, that will be applied for result of last processing function and result of last reduce function.

Reply:
ok